src/app/modules/wallet/pages/account-voluntary-exit/account-voluntary-exit.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, OnInit } from '@angular/core';
+import { BaseComponent } from '../../../shared/components/base.component';
+import { WalletService } from '../../../core/services/wallet.service';
+import { map } from 'rxjs/operators';
+import {
+  Account,
+  AccountVoluntaryExitRequest,
+} from '../../../../proto/validator/accounts/v2/web_api';
+import {
+  FormGroup,
+  FormBuilder,
+  Validators,
+  AbstractControlOptions,
+} from '@angular/forms';
+import { UtilityValidator } from '../../../onboarding/validators/utility.validator';
+import { ActivatedRoute } from '@angular/router';
+import { base64ToHex } from 'src/app/modules/core/utils/hex-util';
+import { MatSelectionListChange } from '@angular/material/list';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+
+@Component({
+  selector: 'app-account-voluntary-exit',
+  templateUrl: './account-voluntary-exit.component.html',
+  styleUrls: ['./account-voluntary-exit.component.scss']
+})
+export class AccountVoluntaryExitComponent   extends BaseComponent
+implements OnInit {
+
+
+  constructor(
+    private walletService: WalletService,
+    private activatedRoute: ActivatedRoute,
+    private snackMsgService: MatSnackBar,
+    private formBuilder: FormBuilder
+  ) {
+    super();
+  }
+  exitAccountFormGroup: FormGroup | undefined | null;
+  keys: Account[] = [];
+  ngOnInit(): void {
+    const publicKey = this.activatedRoute.snapshot.queryParams.publicKey;
```

Comment:
We don't have a `publicKey` query parameter when exiting accounts.

Reply:
Is the idea here that we will use a single public key in the route query parameters to perform the API request? If a user is trying to exit 100 keys at once that might be too big for the route. Is there another way we can pass in a list of public keys we want to exit into this component without using browser query params?

Reply:
This will always be one public key since this navigation is through the grid.
I don't think there is a way to pass the keys to this page any other way since we are navigating to that page.
But we could store them in the service and get them from there